File: Kpd37Gomel/Kpd37Gomel.DataAccess/Models/Vote.cs
﻿using System;
using System.Collections.Generic;

namespace Kpd37Gomel.DataAccess.Models
{
    public class Vote
    {
        public Guid Id { get; set; }

        public Guid AuthorId { get; set; }
        public Tenant Author { get; set; }

        public DateTime CreateDateUtc { get; set; }
        public bool IsPassed { get; set; }

        public string Title { get; set; }
        public string Description { get; set; }
        public bool UseVoteRate { get; set; }
        public bool IsDeleted { get; set; }
        public DateTime? DeletionDateUtc { get; set; }

        public virtual ICollection<VoteVariant> Variants { get; set; } = new List<VoteVariant>();
    }
}

File: Kpd37Gomel/Kpd37Gomel/ClientApp/src/app/vote.ts
import { ITenant } from "./tenant";
import { IVoteVariant } from "./vote-variant";

export interface IVote {
  Id: string;
  Author: ITenant;
  CreateDateUtc: number;
  Title: string;
  Description: string;
  UseVoteRate: boolean;
  IsDeleted: boolean;
  Variants: Array<IVoteVariant>;
}

export class Vote implements IVote {
  constructor(
    public Id: string,
    public Author: ITenant,
    public CreateDateUtc: number,
    public Title: string,
    public Description: string,
    public UseVoteRate: boolean,
    public IsDeleted: boolean,
    public Variants: Array<IVoteVariant>) {}
}
